ibis/backends/tests/tpc/ds/test_queries.py

```diff
@@ -2498,14 +2509,14 @@ def test_49(
             _.channel,
             _.item,
             _.return_ratio,
-            return_rank=_.return_rank + 1,
-            currency_rank=_.currency_rank + 1,
+            return_rank=_.return_rank,
+            currency_rank=_.currency_rank,
         )
         .order_by(
-            _.channel.asc(nulls_first=True),
-            _.return_rank.asc(nulls_first=True),
-            _.currency_rank.asc(nulls_first=True),
-            _.item.asc(nulls_first=True),
+            _[0].asc(nulls_first=True),
+            _[3].asc(nulls_first=True),
+            _[4].asc(nulls_first=True),
+            _[2].asc(nulls_first=True),
```

Comment:
This is causing the non-determinism :)

It should be `_[1]` since we're 0-indexed and SQL column numbers are 1-indexed.

🥲

Reply:
ClickHouse still fails but for a different reason. It seems to compute `rank() over()` differently than everything because of course it does!

Reply:
wow (all of) that's frustrating